src/NHibernate/Type/EnumCharType.cs

```diff
@@ -171,8 +171,6 @@ public override object FromStringValue(string xml)
 		}
 
 		public override string ObjectToSQLString(object value, Dialect.Dialect dialect)
-		{
-			return '\'' + GetValue(value).ToString() + '\'';
-		}
+			=> dialect.ToStringLiteral(GetValue(value).ToString(), SqlType);
```

Comment:
That one is a bit wicked as it does cast the enum underlying integral value as a character, allowing for injecting any character. So, it could abnormally terminate a query by injecting a quote, by example. With a single character injection, it looks like not much harm may be done. But if the query gets batched with others queries, who knows?

So, I have to add a test case for this one.